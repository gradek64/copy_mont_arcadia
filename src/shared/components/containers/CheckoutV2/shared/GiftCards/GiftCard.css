/** @define GiftCard */

.GiftCard {
  padding-top: 10px;

  &-remove {
    text-align: right;
  }

  &-row {
    display: flex;
    justify-content: flex-end;
  }

  &-number {
    margin-right: auto;
    font-size: inherit;
  }

  &-amountUsed {
    color: $success-color;
    text-align: right;
  }
}
