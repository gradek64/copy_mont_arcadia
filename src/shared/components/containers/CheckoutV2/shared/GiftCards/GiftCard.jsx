import PropTypes from 'prop-types'
import React from 'react'

// components
import ButtonLink from '../../../../common/ButtonLink/ButtonLink'
import Price from '../../../../common/Price/Price'

const GiftCard = ({ cardNumber, amountUsed, onRemove }, { l }) => {
  return (
    <div className="GiftCard">
      <div className="GiftCard-row">
        <span className="GiftCard-number">
          {`${l('Gift card number')}: ${cardNumber}`}
        </span>
        <span className="GiftCard-amountUsed">
          -<Price price={amountUsed} />
        </span>
      </div>
      <ButtonLink className="GiftCard-remove" clickHandler={onRemove}>
        {l`Remove this gift card`}
      </ButtonLink>
    </div>
  )
}

GiftCard.propTypes = {
  cardNumber: PropTypes.string.isRequired,
  amountUsed: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  onRemove: PropTypes.func,
}

GiftCard.defaultProps = {
  onRemove: () => {},
}

GiftCard.contextTypes = {
  l: PropTypes.func,
}

export default GiftCard
