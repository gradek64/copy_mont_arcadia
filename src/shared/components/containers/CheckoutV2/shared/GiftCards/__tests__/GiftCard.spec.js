import testComponentHelper from 'test/unit/helpers/test-component'
import GiftCard from '../GiftCard'

describe('<GiftCard />', () => {
  const renderComponent = testComponentHelper(GiftCard)

  const requiredProps = {
    cardNumber: 'XXXX XXXX XXXX 7299',
    amountUsed: '5.00',
  }

  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(requiredProps).getTree()).toMatchSnapshot()
    })
  })

  describe('@events', () => {
    describe('on Remove gift Card button click', () => {
      it('should trigger removeGiftCard action', () => {
        const onRemoveMock = jest.fn()
        const { wrapper } = renderComponent({
          ...requiredProps,
          onRemove: onRemoveMock,
        })
        // call click handler
        const event = {}
        wrapper.find('.GiftCard-remove').prop('clickHandler')(event)
        expect(onRemoveMock).toHaveBeenCalledWith(event)
      })
    })
  })
})
