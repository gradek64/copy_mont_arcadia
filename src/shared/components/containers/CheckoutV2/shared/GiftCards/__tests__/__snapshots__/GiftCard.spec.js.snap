// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiftCard /> @renders in default state 1`] = `
<div
  className="GiftCard"
>
  <div
    className="GiftCard-row"
  >
    <span
      className="GiftCard-number"
    >
      Gift card number: XXXX XXXX XXXX 7299
    </span>
    <span
      className="GiftCard-amountUsed"
    >
      -
      <Price
        className=""
        price="5.00"
      />
    </span>
  </div>
  <ButtonLink
    className="GiftCard-remove"
    clickHandler={[Function]}
    isDisabled={false}
    type="button"
  >
    Remove this gift card
  </ButtonLink>
</div>
`;
